From d1e4ebfd4989298d9fd1098d58a6379de68a310c Mon Sep 17 00:00:00 2001
From: Prathap M S <msprathap@ti.com>
Date: Wed, 30 Oct 2013 20:51:37 +0530
Subject: [PATCH 1/1] Graphics SDK 04.10.00.01 : Change for interfacing with
 SGX DT node

This adds change for interfacing with SGX DT node.
Required for kernel versions >= 3.8 that have SGX DT node integrated.

Signed-off-by: Prathap M S <msprathap@ti.com>
---
 GFX_Linux_KM/services4/srvkm/env/linux/module.c |   19 ++++++++++++++++++-
 1 file changed, 18 insertions(+), 1 deletion(-)

diff --git a/GFX_Linux_KM/services4/srvkm/env/linux/module.c b/GFX_Linux_KM/services4/srvkm/env/linux/module.c
index 668dc2d..2247aad 100644
--- a/GFX_Linux_KM/services4/srvkm/env/linux/module.c
+++ b/GFX_Linux_KM/services4/srvkm/env/linux/module.c
@@ -84,6 +84,9 @@ CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 #include <linux/module.h>
 #include <linux/fs.h>
 #include <linux/proc_fs.h>
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,8,0))
+#include <linux/of.h>
+#endif
 
 #if defined(SUPPORT_DRI_DRM)
 #include <drm/drmP.h>
@@ -296,10 +299,24 @@ static struct platform_device_id powervr_id_table[] __devinitdata = {
 };
 #endif
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,8,0))
+static const struct of_device_id omap_sgx_of_match[] = {
+	{
+		.compatible = "ti,sgx",
+	},
+	{},
+};
+MODULE_DEVICE_TABLE(of, omap_sgx_of_match);
+#endif
+
+
 static LDM_DRV powervr_driver = {
 #if defined(PVR_LDM_PLATFORM_MODULE)
 	.driver = {
 		.name		= DRVNAME,
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,8,0))
+		.of_match_table = of_match_ptr(omap_sgx_of_match),
+#endif
 	},
 #endif
 #if defined(PVR_LDM_PCI_MODULE)
@@ -362,7 +379,7 @@ static int __devinit PVRSRVDriverProbe(LDM_DEV *pDevice, const struct pci_device
 #endif
 {
 	SYS_DATA *psSysData;
-
+	
 	PVR_TRACE(("PVRSRVDriverProbe(pDevice=%p)", pDevice));
 
 #if 0   /* INTEGRATION_POINT */
-- 
1.7.9.5

